core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -742,13 +742,13 @@ protected Releasable getIndexShardOperationsCounter(ShardId shardId) {
         private final List<ShardRouting> shards;
         private final DiscoveryNodes nodes;
         private final boolean executeOnReplica;
-        private final String indexUUID;
         private final AtomicBoolean finished = new AtomicBoolean();
         private final AtomicInteger success = new AtomicInteger(1); // We already wrote into the primary shard
         private final ConcurrentMap<String, Throwable> shardReplicaFailures = ConcurrentCollections.newConcurrentMap();
         private final AtomicInteger pending;
         private final int totalShards;
         private final Releasable indexShardReference;
+        private final ShardRouting primaryShard;
```

Comment:
I merged master into this branch and pushed 16c3862c7604c0a447c96c14a37660196982aa9f.